gpiozero/mixins.py

```diff
@@ -270,13 +280,119 @@ def _fire_events(self):
             if not old_state and new_state:
```

Comment:
Unfortunately that wouldn't work in one rare (but possible) circumstance. Just to help us visualize the code, you're proposing:

```python
if old_state is None:
    # initial state
elif new_state:
    # state change to active
else:
    # state change to inactive
```

Now, consider what happens if the device is currently active and _fire_events is called while the event is still active. It's rare, but in cases of extremely rapid switching it can occur (and different pin implementations make this more or less likely according to their implementation). This is largely because of the way the underlying hardware works. The GPIO registers have bits to set level/edge detection and then a register that indicates whether a level/edge was detected - however that register doesn't tell you *which* edge was detected. In the case you're only looking for a single edge you can obviously figure it out, but in the case you're detecting "both" edges (a common scenario), you can't. Moreover, multiple edges can occur while that register is set (it's just on/off, not a counter), so it's entirely possible to "miss" edges if your code isn't reacting to them fast enough.

So, what do we do in such circumstances? We could report the event anyway - the device (probably) was activated after all, but given that we can't guarantee losing events anyway, it seemed cleaner to only report events on clear transitions between states where we can definitely say we've gone from not active to active and vice versa.

Basically, it's a rather arbitrary design choice - do feel free to open a ticket about it though; it may be worth having a discussion with everyone about what they want.